### PR TITLE
feat: add drag support to alert history horizontal graph

### PR DESCRIPTION
**Pull request by ahmadshaheer**
#### Related Issues / PR's
close https://github.com/SigNoz/signoz/issues/5921

#### Screenshots
https://us04web.zoom.us/clips/share/WeYNubDhLjZvfaX_Rdwh2aQce878hy9gtvjh5MuXaMe3-4V3ysHssXTnihzCLJwD0MqEArOjCpNvDgDaaql8nEdv-w.XQsJltp4tI0MiHtY

### CODE REVIEW

**Comment by YounixM** [comment]
@ahmadshaheer : have we tested that if this affects other areas where we use uPlot ?

**Comment by ahmadshaheer** [comment]
> @ahmadshaheer : have we tested that if this affects other areas where we use uPlot ?

@YounixM, I have made these changes to the timeline graph component of alert history. it is not going to affect any other areas.

**Review by SagarRajput-7 — DISMISSED** [review]

File: frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx [hunk 1]
```diff
@@ -69,6 +76,46 @@ function HorizontalTimelineGraph({
...
+								const startTimestamp = Math.floor(startTime * 1000);
+								const endTimestamp = Math.floor(endTime * 1000);
+
+								if (startTimestamp !== endTimestamp) {
+									dispatch(UpdateTimeInterval('custom', [startTimestamp, endTimestamp]));
+								}
+
+								urlQuery.set(
+									QueryParams.startTime,
+									Math.floor(startTime * 1000).toString(),
```
Comment (SagarRajput-7) [inline frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx:103]: can use startTimestamp and endTimestamp variables here

**Review by vikrantgupta25 — DISMISSED** [review]

File: frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx (hunk 1 above)
Comment (ahmadshaheer) [reply frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx:103]: Done

File: frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx [hunk 2]
```diff
@@ -66,6 +73,40 @@ function HorizontalTimelineGraph({
 					label: 'States',
 				},
 			],
+			hooks: {
+				setSelect: [
+					(self): void => {
+						const selection = self.select;
+						if (selection) {
+							const startTime = self.posToVal(selection.left, 'x');
+							const endTime = self.posToVal(selection.left + selection.width, 'x');
```
Comment (vikrantgupta25) [inline frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx:82]: i hope we have handled the back navigation as well ? example -> if we move from timerange A to B then on moving back the time is correctly updated ?

Comment (ahmadshaheer) [reply frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx:82]: Thanks for pointing this out, @vikrantgupta25, I fixed this issue :slightly_smiling_face:

**Review by YounixM — APPROVED** [review]

File: frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahmadshaheer) [reply frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx]: I had already made these changes, @rahulkeswani101 :slightly_smiling_face: .

[the remark this answers is not in the document]
Comment (rahulkeswani101) [reply frontend/src/container/AlertHistory/Timeline/Graph/Graph.tsx]: @ahmadshaheer Added this comment 5 days ago.

**Pull request merged**